osu.Game/OsuGame.cs

```diff
@@ -1033,6 +1035,49 @@ private void forwardLoggedErrorsToNotifications()
             };
         }
 
+        private void forwardTabletLogsToNotifications()
+        {
+            bool notifyOnWarning = true;
+
+            Logger.NewEntry += entry =>
+            {
+                if (entry.Level < LogLevel.Important || entry.LoggerName != ITabletHandler.LOGGER_NAME)
+                    return;
+
+                if (entry.Level == LogLevel.Error)
+                {
+                    Schedule(() => Notifications.Post(new SimpleNotification
+                    {
+                        Text = $"Encountered tablet error: \"{entry.Message}\"",
+                        Icon = FontAwesome.Solid.PenSquare,
+                        IconColour = Colours.RedDark,
+                    }));
+                }
+                else if (notifyOnWarning)
+                {
+                    Schedule(() => Notifications.Post(new SimpleNotification
+                    {
+                        Text = @"Encountered tablet warning, your tablet may not function correctly. Click here for a list of all tablets supported.",
+                        Icon = FontAwesome.Solid.PenSquare,
+                        IconColour = Colours.YellowDark,
+                        Activated = () =>
+                        {
+                            OpenUrlExternally("https://opentabletdriver.net/Tablets", true);
+                            return true;
+                        }
+                    }));
+
+                    notifyOnWarning = false;
+                }
+            };
+
+            Schedule(() =>
+            {
+                ITabletHandler tablet = Host.AvailableInputHandlers.OfType<ITabletHandler>().SingleOrDefault();
+                tablet?.Tablet.BindValueChanged(_ => notifyOnWarning = true, true);
+            });
```

Comment:
This is quite ugly, but is the best way to:
 - keep this method self-contained and allow for splitting (by not moving `notfyOnWarning` to a field and bind tablet bindable when game loaded),
 - and also catch any tablet logs that were reported prior to the game being loaded (rather than calling this method once game is loaded, at which point warning/error from a connected tablet may not be visible).

The point of this logic is to keep notifying the user on warning when switching tablets (albeit the notification doesn't clearly state which tablet raised it). This could alternatively be resolved by splitting this to a `Component` that binds to the event on constructor and binds to tablet bindable on BDL.

Reply:
Based on the comment above, I'm still not sure what this code is doing in the first place. Is it a debounce? For what?